utils/writeDataFile.js

```diff
@@ -0,0 +1,31 @@
+const fs = require('fs');
+
+// iterates over a mongocursor and writes a json file with "data" and "metadata" key-value pairs
+const writeDataFile = (fileName, mongoCursor, metadata, callback) => {
+  // init writable stream
+  const file = fs.createWriteStream(fileName, { flags: 'a' });
+
+  // create opening lines of json file
+  file.write('{\n"data": {"actions":[\n');
+
```

Comment:
Maybe you should set an error handler:
`file.on('error', err => { throw err; })`

The outer .catch() will catch this.

Reply:
👍

Reply:
👍